src/PollingBlockTracker.test.ts

```diff
@@ -834,6 +834,262 @@ describe('PollingBlockTracker', () => {
         },
       );
     });
+
+    describe('after a block number is cached', () => {
```

Comment:
Given that we also have a `describe` block "after a block number is cached if the block tracker was initialized with \`usePastBlocks: true\`", there must be a reason this one exists, and it seems that in this case the difference is that `usePastBlocks` is _not_ being given. Also, I feel like we need to explicitly test what happens when `usePastBlocks` is false (since that's also a possibility) — though in that case we'd have the same expectation as if it were not given at all.

Putting this together, what do you think about using a loop that looks something like this?

``` typescript
describe.each([
 [
   'initialized with `usePastBlocks: true`',
   { usePastBlocks: true }
 ],
 ['not initialized with `usePastBlocks`', {}],
] as const)(
  'after a block number is cached if the block tracker was %s',
  (_description, blockTrackerOptions) => {
    // ... tests go here ...
  });
);
```

And inside the tests, when you call `withPollingBlockTracker`, you could have something like:

``` javascript
await withPollingBlockTracker(
  {
    provider: {
      // ...
    },
    blockTracker: blockTrackerOptions,
  },
  // ...
)
```

The same thing applies to all of the other instances where you have this kind of `describe` block.

Reply:
I tried cleaning the tests up and think it's a bit better now with the loops, but I'm wary about going too much further. The tests may have similar shape, but they're slightly different enough that it was a pain to DRY up. I'm a little worried we've introduced too much complexity in the tests now.

Reply:
Hmm, I feel like I may have miscommunicated. I didn't intend for all of the tests to be DRYed up, just the ones that are exactly the same (when `usePastBlocks` is false and when `usePastBlocks` is not given, since they should have the same behavior).

I don't want you to have to do a lot more work on this PR since you've done so much already, so maybe to fix this in an easier fashion, you could revert your latest commit, then replace

```
describe('after a block number is cached', () => {
```

with

```
describe('after a block number is cached if the block tracker was initialized with usePastBlocks: false', () => {
```

and then copy that whole block, but then rename the describe to

```
describe('after a block number is cached if the block tracker was not initialized with usePastBlocks', () => {
```

Does that make sense? Sorry I wasn't more clear :(

Reply:
This was my bad. I think what threw me off was that your previous comment was on the line 
`describe('after a block number is cached', () => {` which made me believe that you wanted not only `{ usePastBlocks: true }` and `{ usePastBlocks: false }` but also `{}` in the same loop. I understand now what you were asking for.

I should have checked in again to clarify what you were asking for once it was clear that this was going to introduce a lot more complexity to the tests. Sorry about that.

Reply:
Oh... you're totally right, sigh. I gave you `usePastBlocks: true` when I should have said `usePastBlocks: false`. Alright, well, I'm glad it wasn't too bad to undo that work.